website/docs/bestpractices/analytics/trino-best-practices.md

```diff
@@ -0,0 +1,642 @@
+
+#### Configuring Keda
+- Add Keda helm release to the cluster
+- Add JVM / JMX Exporter configuration to Trino Helm value, enable serviceMonitor
+```
+configProperties: |-
+      hostPort: localhost:{{- .Values.jmx.registryPort }}
+      startDelaySeconds: 0
+      ssl: false
+      lowercaseOutputName: false
+      lowercaseOutputLabelNames: false
+      whitelistObjectNames: ["trino.execution:name=QueryManager","trino.execution:name=SqlTaskManager","trino.execution.executor:name=TaskExecutor","trino.memory:name=ClusterMemoryManager","java.lang:type=Runtime","trino.memory:type=ClusterMemoryPool,name=general","java.lang:type=Memory","trino.memory:type=MemoryPool,name=general"]
+      autoExcludeObjectNameAttributes: true
+      excludeObjectNameAttributes:
+        "java.lang:type=OperatingSystem":
+          - "ObjectName"
+        "java.lang:type=Runtime":
+          - "ClassPath"
+          - "SystemProperties"
+      rules:
+      - pattern: ".*"
+```
+- Deploy KEDA scaledObject for trino, tracking CPU and QueuedQueries. The 'target' CPU percentage here is set at 85%, to leverage the physical cores of Graviton instances.
+```
+triggers:
+  - type: cpu
+    metricType: Utilization
+    metadata:
+      value: '85'  # Target CPU utilization percentage
+  - type: prometheus
+    metricType: Value
+    metadata:
+      serverAddress: http://kube-prometheus-stack-prometheus.kube-prometheus-stack.svc.cluster.local:9090
+      threshold: '1'
+      metricName: queued_queries
+      query: sum by (job) (avg_over_time(trino_execution_QueryManager_QueuedQueries{job="trino"}[1m]))
+```
+
+## File Caching
+
+File caching provides three major benefits to storage systems and query operations. First, it reduces the storage load by preventing repetitive retrievals of the same files, as cached files can be reused across multiple queries on the same worker. Second, it can significantly improve query performance by eliminating repeated network transfers and allowing access to local file copies, particularly beneficial when the original storage is in a different network or region. Finally, it leads to reduced query costs by minimizing network traffic and storage access.
+
+This is a basic configuration for implementing File Caching.
+
+```
+fs.cache.enabled=true
+fs.cache.directories=/tmp/cache/
+fs.cache.preferred-hosts-count=10 # The cluster size determines the host count. We recommend keeping host count small to maintain optimal performance.
+```
+
+:::note
+Configure instances with Local SSD storage for cache directories to significantly improve I/O speed and overall query performance.
+:::
+
+## Trino Gateway
```

Comment:
I would suggest removing this if we're not adding any content under it. We can add it back when we have populated the section.

Reply:
Done